### PR TITLE
Bug 1716951 - Implement the Custom Distribution Metric

### PR DESCRIPTION
**Pull request by rosahbruno**
Implements the Custom Distribution metric.

The documentation for the Glean book is updated in this PR - https://github.com/mozilla/glean/pull/2205

### Pull Request checklist ###

- [x] **Quality**: Make sure this PR builds and runs cleanly.
  - Inside the `glean/` folder, run:
    - `npm run test` Runs _all_ tests
    - `npm run lint` Runs _all_ linters
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
- [x] **Documentation**: This PR includes documentation changes, an explanation of why it does not need that or a follow-up bug has been filed to do that work
  - Documentation should be added to [The Glean Book](https://mozilla.github.io/glean/book/index.html) when making changes to Glean's user facing API
    - When changes to the Glean Book are necessary, link to the corresponding PR on the [`mozilla/glean`](https://github.com/mozilla/glean) repository
  - Documentation should be added to [the Glean.js developer documentation](https://github.com/mozilla/glean.js/tree/main/docs) when making internal code changes

### CODE REVIEW

File: glean/tests/unit/core/metrics/custom_distribution_exponential.spec.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    sandbox.restore();
+  });
+
+  it("serializer should correctly serialize custom distribution", async function () {
+    const metric = new CustomDistributionMetricType(
+      {
+        category: "aCategory",
+        disabled: false,
+        lifetime: Lifetime.Ping,
+        name: "aMemoryDistribution",
```
Comment (Dexterp37) [inline glean/tests/unit/core/metrics/custom_distribution_exponential.spec.ts:32]: nit: Not a big deal, but this is not a memory distribution :)

File: glean/tests/unit/core/metrics/custom_distribution_exponential.spec.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+  it("set value properly sets the value in all stores", function () {
+    const storesNames = ["aPing", "bPing"];
+
+    const metric = new CustomDistributionMetricType(
+      {
+        category: "aCategory",
+        disabled: false,
+        lifetime: Lifetime.Ping,
+        name: "aMemoryDistribution",
```
Comment (Dexterp37) [inline glean/tests/unit/core/metrics/custom_distribution_exponential.spec.ts:55]: nit: Not a big deal, but this is not a memory distribution :)

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
only a few nits! Good job.

File: glean/tests/unit/core/metrics/custom_distribution_exponential.spec.ts (hunk 1 above)
Comment (rosahbruno) [reply glean/tests/unit/core/metrics/custom_distribution_exponential.spec.ts:32]: omg, the problem of when you just keep copying and pasting the same metric haha, will fix

**Review by Dexterp37 — APPROVED** [review]
Thank you!

**Pull request merged**